rfcs/accessibility-checklist.md

```diff
@@ -0,0 +1,52 @@
+
+
+https://user-images.githubusercontent.com/35374649/175493627-4c5f29e9-0625-44fe-a5a8-03df7700f886.mp4
+
+
+## Component Decisions
+
+While writing `decision.md` file for component make sure to include `Accessibility` section and link to the appropriate [WAI ARIA component pattern](https://www.w3.org/WAI/ARIA/apg/patterns/) 
+
+Example: 
+
+Let's say you are building a `Checkbox` component. 
+
+- Step 1: Write the core logic for checkbox behaviour 
+- Step 2: Find the appropriate [WAI-ARIA Pattern for checkbox](https://www.w3.org/WAI/ARIA/apg/patterns/checkbox/)
+- Step 3: Accordion to the pattern implement [proper keyboard accessibility](https://www.w3.org/WAI/ARIA/apg/patterns/checkbox/#keyboard-interaction-5), like in checkbox we have to make sure to pressing <kbd>Space</kbd> should toggle the state. 
+- Step 4: Each pattern will have [an example section](https://www.w3.org/WAI/ARIA/apg/patterns/checkbox/#examples-1) which you can check for implementation and detailed information about [Accessibility Features](https://www.w3.org/WAI/ARIA/apg/example-index/checkbox/checkbox.html#accessibilityfeatures), [Keyboard Support](https://www.w3.org/WAI/ARIA/apg/example-index/checkbox/checkbox.html#kbd_label) and what various [aria attributes](https://www.w3.org/WAI/ARIA/apg/example-index/checkbox/checkbox.html#rps_label) the component needs to be accessible. 
+  
+Following these patterns & examples we can make our components accessible.  
+
+## Implementations
+
+To implement the proper aria attributes we can make use of our `makeAccessible` function which we already [discussed in the RFC](https://github.com/razorpay/blade/blob/master/rfcs/2022-04-09-accessibility.md#platform-specific-implementation--5). 
+
+At this point we will also adhere to all the keyboard specific accessibility features, and implement proper keyboard management per component basis. 
+
+## Manual Testing & Review
+
+After building the component we have to do few rounds of manual testing to make sure that screen readers are announcing the component correctly.
+
+In MacOS we can press `fn+⌘+f5` to open up voice-over and test out our component with keyboard only. 
+
+We also have a manual testing checklist in [Accessibility RFC](https://github.com/razorpay/blade/blob/master/rfcs/2022-04-09-accessibility.md#manual-testing)
+
+> If you are not familar with using screen reader you can [follow this guide](https://dequeuniversity.com/tips/learn-voiceover),
+> And also ask for manual reviews from Anurag or Abinash.
+
+## Checklist
+
+- [ ] Link appropriate component pattern in `decisions.md` file
+- [ ] Read and learn about the component's accessibility requirements
+- [ ] Checkout examples for the component
+- [ ] Adhere to the aria states & roles for the component
+- [ ] Adhere to the keyboard accessibility requirements for the component
+- [ ] Do atleast one manual testing with voice over
+- [ ] Take accessibility reviews from other team members
```

Comment:
the list seems very overwhelming to understand for someone who has less context about a11y. Proposing my version below


```
## Checklist
- [ ] Add `Accessibility` section to component's `decision.md`. [More Details](link to the section)
- [ ] Use `makeAccessible` utility to make components accessible across platforms. [More Details](link to the section)
- [ ] Adhere to the aria states & roles for the component
- [ ] Adhere to the keyboard accessibility requirements for the component
- [ ] Manual Testing and Self Review
	- [ ] Interactive elements in the page Navigable through the keyboard ([Success Criterion](https://www.w3.org/WAI/WCAG21/Techniques/general/G202))
	- [ ] Actions can be triggered by keyboard ([Failure of Success Criterion](https://www.w3.org/WAI/WCAG21/Techniques/failures/F54.html))
	- [ ] Interactive elements remain focused until the user moves it ([Failure of Success Criterion](https://www.w3.org/WAI/WCAG21/Techniques/failures/F55.html))
	- [ ] The tab order match the logical reading order of the page ([Failure of Success Criterion](https://www.w3.org/WAI/WCAG21/Techniques/failures/F44.html))
	- [ ] All the keyboard interactive elements display visual keyboard focus ([Failure Of Success Criterion](https://www.w3.org/WAI/WCAG21/Techniques/failures/F78.html))
	- [ ] The visual keyboard focus is easy to identify ([Failure Of Success Criterion](https://www.w3.org/WAI/WCAG21/Techniques/failures/F78.html))
	- [ ] During the navigation, you don't become trapped in an element ([Failure of Success Criterion](https://www.w3.org/WAI/WCAG21/Techniques/failures/F10.html))
	- [ ] Hidden popuops or titles are accessible through keyboard.
- [ ] Take accessibility reviews from other team members(whenever required)

```

Reply:
This looks great, I've removed the Manual testing items, They are mostly page wide accessibility checklist would not make sense to add it to component level checklist so removed them to simplify things.